### PR TITLE
more arguments for prepare

### PR DESCRIPTION
**Pull request by thebookins**
To use the `prepare` step in an `openaps` workflow, and properly resolve pump suspends, the current clock time and duration of the history window need to be passed in as arguments and routed to the `clean` step. This PR adds options `--end`, `--start` and `--duration` to the interface for `prepare`. For consistency, the `clean` step is also modified to match `trim` and `prepare` and will now accept start and end times in JSON format. Needed for https://github.com/openaps/oref0/pull/279.

### CODE REVIEW

File: openapscontrib/mmhistorytools/historytools.py [hunk 1]
```diff
@@ -114,6 +114,11 @@ def __init__(self, trimmed_history, start_datetime=None, end_datetime=None):
         self.end_datetime = end_datetime
 
         if len(trimmed_history) > 0:
+            if start_datetime is None and end_datetime is not None and duration_hours is not None:
```
Comment (loudnate) [inline openapscontrib/mmhistorytools/historytools.py]: Would it be worthwhile to add test coverage for the duration argument? You should be able to copy/adapt the existing tests for duration.

Comment (thebookins) [inline openapscontrib/mmhistorytools/historytools.py]: Good idea, I'll add test coverage.

**Comment by thebookins** [comment]
I've added test coverage for the duration argument, and found a mistake in the process (now fixed). Thanks @loudnate for your feedback, and for building this great set of tools in the first place.

**Comment by thebookins** [comment]
Thanks. Not expecting to make any more changes, so would be great if you could go ahead with the release.

**Pull request merged**